### PR TITLE
add FinalEntityState to context

### PR DESCRIPTION
**Pull request by galkoren**

### CODE REVIEW

File: main/src/main/java/com/kenshoo/pl/entity/FinalEntityState.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package com.kenshoo.pl.entity;
+
+import java.util.List;
+
+public class FinalEntityState implements Entity {
+
+    final CurrentEntityState currentState;
+    final EntityChange<? extends EntityType<?>> change;
```
Comment (effiban) [inline main/src/main/java/com/kenshoo/pl/entity/FinalEntityState.java:8]: The above 2 should be `private`

File: main/src/main/java/com/kenshoo/pl/entity/FinalEntityState.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
+package com.kenshoo.pl.entity;
+
+import java.util.List;
+
+public class FinalEntityState implements Entity {
```
Comment (dimagorelik) [inline main/src/main/java/com/kenshoo/pl/entity/FinalEntityState.java:5]: @galkoren 
We already have a similar class. Please, see ResultingFieldsCombination

Comment (effiban) [reply main/src/main/java/com/kenshoo/pl/entity/FinalEntityState.java:5]: @dimagorelik - there are significant differences between them:
1) `ResultingFieldsCombination` is only relevant for fields of one entity type (same type as the command) but we need to support the more general case of fields from anywhere in the hierarchy
2) `ResultingFieldsCombination` filters by a stream of fields in the constructor and we don't want that limitation either
3) `ResultingFieldsCombination` implements `FieldsValueMap` and not `Entity` so it doesn't match the pattern we want of current entity and final entity implementing the same interface.

So it seems like `ResultingFieldsCombination` is a special case of `FinalyEntityState` (more or less) and I think we can leave it for a separate PR to see if and how we can combine them.

Comment (galkoren) [reply main/src/main/java/com/kenshoo/pl/entity/FinalEntityState.java:5]: I can't use it. The "Field Combination" is for a set of fields of a virtual field.
I don't have a set of fields.
The FinalState will soon calculate the complicated merging of fetchMany, which is not something the "Field Combination" could do.

Comment (dimagorelik) [reply main/src/main/java/com/kenshoo/pl/entity/FinalEntityState.java:5]: @galkoren 
I agree but maybe the new final state class can simplify the validator interface

**Comment by effiban** [comment]
@galkoren - please add unit tests

**Pull request merged**